external-crates/move/crates/move-compiler/src/expansion/aliases.rs

```diff
@@ -205,78 +293,117 @@ impl AliasMapBuilder {
         alias: Name,
         ident: ModuleIdent,
         member: Name,
+        kind: ModuleMemberKind,
     ) -> Result<(), Loc> {
-        let result = self.remove_member_alias_(&alias);
-        self.members
-            .add(alias, ((ident, member), /* is_implicit */ true))
-            .unwrap();
-        result
+        self.add_member_alias_(alias, ident, member, kind, /* is_implicit */ true)
     }
 }
 
+macro_rules! resolve_alias {
+    ($map:expr, .$namespace:ident, $name:expr) => {{
+        let mut current_scope = Some($map);
+        let name = $name;
+        loop {
+            let Some(scope) = current_scope else {
+                break None;
+            };
+            let Some(entry) = scope.$namespace.get(name).copied() else {
+                current_scope = scope.previous.as_mut().map(|x| &mut **x);
+                continue;
+            };
+            // Note, might have already been removed by a different `$namespace` resolution
+            scope.unused.remove(&(*name, entry).into());
```

Comment:
Is there an advantage to tracking `unused` separately and duplicating the alias entries into the set, versus a boolean flag on the entries themselves (which avoids the extra lookup)?

Reply:
I just feel like this is a bit less brittle. Some aliases are spread across the different namespaces, and it felt strange for just types to have to go lookup into another set. 

In other words, this helps isolate the knowledge about what goes into what namespace into just the creation... I hope at least

Reply:
Though I suppose that isn't really true because of the removal?

Reply:
It at the very least isolates it to a single file (after I move this to its own file)